react/components/molecules/tabs-text/tabs-text.js

```diff
@@ -52,10 +54,13 @@ export class TabsText extends PureComponent {
     }
 
     onTabPress = tabSelectedIndex => {
-        this.setState({ tabSelected: tabSelectedIndex }, () => {
-            this._updateBar();
-            this.props.onTabChange(this.state.tabSelected);
-        });
+        if (this.state.tabSelected === tabSelectedIndex) this.props.onSelectedTabPress();

```

Comment:
Go with "early exit" strategy

```suggestion
        if (this.state.tabSelected === tabSelectedIndex) {
            this.props.onSelectedTabPress();
            return;
        }
```

Reply:
(and you can delete the "else")